modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,728 @@
+			if ( $attachment_count[ $attachment->post_parent ] >= 5 ) {
+				continue;
+			}
+
+			$selected_attachments[] = $attachment->ID;
+			$attachment_count[ $attachment->post_parent ] ++;
+		}
+
+		// bail if there weren't any attachments to avoid an extra query
+		if ( empty( $selected_attachments ) ) {
+			continue;
+		}
+
+		// Get more of the attachment object for the attachments we actually care about
+		$attachment_ids   = implode( ',', array_map( 'intval', $selected_attachments ) );
+		$query            = "SELECT p.ID, p.post_parent, p.post_title, p.post_excerpt, p.guid FROM {$wpdb->posts} as p WHERE p.ID IN ({$attachment_ids}) AND p.post_type='attachment' AND p.post_mime_type='image/jpeg' LIMIT 500;";
+		$attachments      = $wpdb->get_results( $query );
+		$post_attachments = array_merge( $post_attachments, $attachments );
+	}
+
+	unset( $initstr );
+	$latest_mod = '';
+	foreach ( $posts as $post ) {
+
+		/**
+		 * Filter condition to allow skipping specific posts in sitemap.
+		 *
+		 * @module xml-sitemap
+		 *
+		 * @since 3.9
+		 *
+		 * @param bool $skip Current boolean. False by default, so no post is skipped.
+		 * @param WP_POST $post Current post object.
+		 */
+		if ( apply_filters( 'jetpack_sitemap_skip_post', false, $post ) ) {
+			continue;
+		}
+
+		$post_latest_mod = null;
+		$url             = array ( 'loc' => esc_url( get_permalink( $post->ID ) ) );
+
+		// If this post is configured to be the site home, skip since it's added separately later
+		if ( untrailingslashit( get_permalink( $post->ID ) ) == untrailingslashit( get_option( 'home' ) ) ) {
+			continue;
+		}
+
+		// Mobile node specified in http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
+		$url['mobile:mobile'] = '';
+
+		// Image node specified in http://support.google.com/webmasters/bin/answer.py?hl=en&answer=178636
+		// These attachments were produced with batch SQL earlier in the script
+		if ( ! post_password_required( $post->ID ) && $attachments = wp_filter_object_list( $post_attachments, array ( 'post_parent' => $post->ID ) ) ) {
+
+			$url['image:image'] = array ();
+
+			foreach ( $attachments as $attachment ) {
+				$attachment_url = false;
+				if ( $attachment->guid ) {
+
+					// Copied from core's wp_get_attachment_url(). We already
+					// have the guid value, so we don't want to get it again.
+					// Note: we're using the WP.com version of the function.
+
+					/** This filter is documented in wp-includes/post-template.php */
+					$attachment_url = apply_filters( 'get_the_guid', $attachment->guid );
+
+					/** This filter is documented in wp-includes/post.php */
+					$attachment_url = apply_filters( 'wp_get_attachment_url', $attachment_url, $attachment->ID );
+
+					// If we don't have an attachment URL, don't include this image
+					if ( ! $attachment_url ) {
+						unset( $url['image:image'] );
+						continue;
+					}
+
+					$url['image:image']['loc'] = esc_url( $attachment_url );
```

Comment:
I didn't fully trace this back. I don't think this would pick up a Photonized version of an image if Photon is active. Should we use Photon for this or stick to the original?